sui-execution/latest/sui-adapter/src/static_programmable_transactions/metering/loading.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::static_programmable_transactions::{
+    loading::ast as L, metering::translation_meter::TranslationMeter,
+};
+use sui_types::error::ExecutionError;
+
+/// After loading and before type checking we do a pass over the loaded transaction to charge for
+/// types that occured in the transaction and were loaded. We simply charge for the number of type
+/// nodes that were loaded.
+pub fn meter(
+    meter: &mut TranslationMeter,
+    transaction: &L::Transaction,
+) -> Result<(), ExecutionError> {
+    let inputs = transaction.inputs.iter().filter_map(|i| match &i.1 {
+        L::InputType::Bytes => None,
+        L::InputType::Fixed(ty) => Some(ty),
+    });
+    let commands = transaction.commands.iter().flat_map(command_types);
+    for ty in inputs.chain(commands) {
+        meter.charge_num_type_nodes(ty.node_count())?;
+    }
+    Ok(())
+}
+
+fn command_types(cmd: &L::Command) -> Box<dyn Iterator<Item = &L::Type> + '_> {
```

Comment:
I always forget I can `Box::dyn` here... I usually reach for itertools `Either` but this is definitely cleaner